wikimedia/wikipedia-ios
# Drop iOS 14, tweaks for Xcode 15

**Phabricator:** 
https://phabricator.wikimedia.org/T343038
https://phabricator.wikimedia.org/T344005

### Notes
These are changes needed to drop support for iOS 14. This is best reviewed commit-by-commit.

A lot of warnings appeared after bumping the deployment target. I think I fixed 27a2bdf properly. The other two commits I described as suppressions, as they felt like more of a workaround rather than actually fixing. Most of the warnings were in a0a4837 - in my mind properly fixing these buttons with `UIButtonConfiguration` would have been a significant amount of rework that we already want to do with Components rewrites, so I resorted to some warning suppressions. Happy to discuss that further if there's disagreement on this though.

The project mostly already worked in Xcode 15, so there are some minor commits towards the end for that. I think Xcode 14.3.1 should still run fine with these (which our PR unit tests are still set against). I also kicked off an Experimental build against Xcode 15 beta to confirm deployment will work.

### Test Steps
1. Just general regression. Run on simulators, test devices, in both iOS 15, 16, and 17. Ensure no warnings appear.
